docs/osquery/osquery-placeholder-fields.asciidoc

```diff
@@ -0,0 +1,30 @@
+[[osquery-placeholder-fields]]
+== Use placeholder fields in Osquery queries 
+
+Instead of hard-coding alert and event values into Osquery queries, you can use placeholder fields to dynamically pass this data into queries. Placeholder fields function like replaceable parameters. You can use placeholder fields to build flexible and re-usable queries. 
+
+Placeholder fields work in single queries or query packs. They are also supported in the following features:
+
+* <<alerts-run-osquery,Live queries>>
+* <<osquery-response-action,Osquery Response Action>>
++
+NOTE: Placeholder fields within an Osquery Response Action query will only work if the rule has generated alerts.
+* <<invest-guide-run-osquery,Investigation guides using Osquery queries>> 
+
+[float]
+[[placeholder-field-syntax]]
+=== Placeholder field syntax and requirements
+
+Placeholder fields use http://mustache.github.io/[mustache syntax] and must be wrapped in double curly brackets (`{{}}`). You can use any field within an event or alert document as a placeholder field. 
+
+Queries with placeholder fields _must_ be run against an alert or event. If a placeholder field can't find an alert or event value, the query status will be `Skipped`. 
```

Comment:
```suggestion
Queries with placeholder fields can only run against alerts or events. Otherwise, they will lack the necessary values and be `Skipped`. 
```

Reply:
@benironside I'm fairly certain that the second sentence needs to be kept as is. If an Osquery query is using a placeholder field to retrieve a field that doesn't exist on the alert or event, the query will fail regardless of where it was ran against. @szwarckonrad @tomsonpl can you verify what I said is correct? 

@szwarckonrad @tomsonpl - On a related note, during my testing, I've seen the `error` status returned for failed queries (i.e. queries using placeholder fields to pass in values for fields that don't exist on an alert or event). Is this correct? Should I replace `Skipped` with `error`?

Reply:
Hey @nastasha-solomon , @benironside - to me both of your version seem the same (correct) :P 

Whenever we use the {{ data }} dynamic params syntax within context of Alert or Event (we need its data) - we try to replace the parameter with the data, and if we fail (eg. there is no data held under the key we provided) - we return an `error` saying "This query hasn't been called due to parameter used and its value not found in the alert." which means that in the end we didn't run this osquery on the host (not sending action to fleet). To me this can safely be referred to as something like "The query was skipped" 

I am sorry as this might sound confusing, but I'll be happy to discuss this further if that would help 👍

Reply:
Ah, ok thanks for the additional information about the error message and how Fleet perceives (or doesn't perceive?) the "failed" query. With this in mind, I agree that @benironside's edit makes sense to use. Thanks, both!